behavior/MouseTracking/get_mouse_XY_pos.m

```diff
@@ -112,14 +126,33 @@
             end
 
             % Save centroids data and update plot
+            true_frame_idx = frame_chunks(idx,1)+frame_idx-1;           
+                
             if isempty(area_vector) %sometimes blob disappears, go to previous centroid
-                centroids(frame_chunks(idx,1)+frame_idx-1,:) = c_old;
-                centroid_color = 'r';
+                
+                % trial_aware mode
+                % if image is the first image of a trial, use next centroid
+                if trial_aware && ~isempty(find(true_frame_idx==trial_indices(:,1),1))
```

Comment:
This can be written simply (but double check):
```
if trial_aware && find(true_frame_idx==trial_indices(:,1),1)
```
since `find` will return an empty list (`[]`) when there is no match; and the empty list is equivalent to a false in Matlab.